store/mongo/datastore_mongo.go

```diff
@@ -654,15 +655,23 @@ func (db *DataStoreMongo) UpsertRemoveDeviceAttributes(
 		update["$unset"] = remove
 	}
 
-	var res *mongo.UpdateResult
-	res, err = c.UpdateOne(ctx, filter, update, updateOpts)
-	if err == nil {
-		result = &model.UpdateResult{
-			MatchedCount: res.MatchedCount,
-			CreatedCount: res.UpsertedCount,
-		}
+	device := &model.Device{}
+	res := c.FindOneAndUpdate(ctx, filter, update, updateOpts)
```

Comment:
I don't get why we need to return the device document. Isn't this just extra overhead?

Reply:
We need to return the document to (eventually) reindex the full-text field.

It is technically possible to infer the full-text field without retrieving the document after the update, but this can lead to race conditions and we wanted to come up with a simple solution, at least at this stage. Therefore, we update inventory attributes/tags retrieving the document after the update, compose the full-text field and update it with an independent query if different from the one we retrieved from the db.